ecs/pod_creator_test.go

```diff
@@ -1,12 +1,286 @@
 package ecs
 
 import (
+	"context"
 	"testing"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/evergreen-ci/cocoa"
+	"github.com/evergreen-ci/cocoa/internal/awsutil"
+	"github.com/evergreen-ci/cocoa/internal/testutil"
+	"github.com/evergreen-ci/cocoa/secret"
+	"github.com/evergreen-ci/utility"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestECSPodCreatorInterface(t *testing.T) {
 	assert.Implements(t, (*cocoa.ECSPodCreator)(nil), &BasicECSPodCreator{})
 }
+
+func TestECSPodCreator(t *testing.T) {
+	testutil.CheckAWSEnvVarsForECSAndSecretsManager(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault){
+		"NewPodCreatorFailsWithMissingClient": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			podCreator, err := NewBasicECSPodCreator(nil, nil)
+			require.Error(t, err)
+			require.Zero(t, podCreator)
+		},
+		"CreatePodFailsWithInvalidCreationOpts": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			opts := cocoa.NewECSPodCreationOptions()
+			require.NotZero(t, opts)
+			assert.Zero(t, *opts)
+
+			podCreator, err := NewBasicECSPodCreator(c, nil)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithSecretsNoVault": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("evName").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName("name").SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar})
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
```

Comment:
This test should set the execution role to ensure that the test doesn't just fail because the execution role is missing.